cmd/minikube/cmd/service.go

```diff
@@ -159,22 +159,22 @@ You may select another namespace by using 'minikube service {{.service}} -n <nam
 			}
 		}
 
-		noNodePortSvcNames := []string{}
-		for _, svc := range noNodePortServices {
-			noNodePortSvcNames = append(noNodePortSvcNames, fmt.Sprintf("%s/%s", svc.Namespace, svc.Name))
-		}
-		if len(noNodePortServices) != 0 {
-			out.WarningT("Services {{.svc_names}} have type \"ClusterIP\" not meant to be exposed, however for local development minikube allows you to access this !", out.V{"svc_names": noNodePortSvcNames})
-		}
-
-		if driver.NeedsPortForward(co.Config.Driver) && services != nil {
-			startKicServiceTunnel(services, cname, co.Config.Driver)
+		if driver.NeedsPortForward(co.Config.Driver) {
+			svcs := services
+			if len(svcs) == 0 && len(noNodePortServices) > 0 {
+				svcs = noNodePortServices
+
+				noNodePortSvcNames := []string{}
+				for _, svc := range noNodePortServices {
+					noNodePortSvcNames = append(noNodePortSvcNames, fmt.Sprintf("%s/%s", svc.Namespace, svc.Name))
+				}
+				out.WarningT("Services {{.svc_names}} have type \"ClusterIP\" not meant to be exposed, however for local development minikube allows you to access this !", out.V{"svc_names": noNodePortSvcNames})
```

Comment:
thank you for this PR, I think @prezha this warning should be printed for all drivers, no ?
the idea was educatiing the users that clusterIP services are not meant to be exposed, and this is an exception for learning DONT get used to it